### PR TITLE
Post image

### PR DESCRIPTION
**Pull request by tux255**
Post images and tests updated
Post trix editor

### CODE REVIEW

File: app/helpers/image_helper.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# frozen_string_literal: true
+module ImageHelper
```
Comment (igor04) [inline app/helpers/image_helper.rb:2]: no need in blank helpers

File: app/models/category.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+class Category < ApplicationRecord
+  belongs_to :user
+
+  has_many :subcategories, class_name: 'Category', foreign_key: 'parent_id', dependent: :destroy
+  belongs_to :parent_category, class_name: 'Category', optional: true
+  has_many :subcategory_posts, through: :subcategories, source: :posts
+
+  has_many :posts, foreign_key: 'category_id'
```
Comment (igor04) [inline app/models/category.rb:9]: https://github.com/tux255/sports_hub_portal/pull/7#discussion_r658520775

File: app/models/category.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+class Category < ApplicationRecord
+  belongs_to :user
+
+  has_many :subcategories, class_name: 'Category', foreign_key: 'parent_id', dependent: :destroy
+  belongs_to :parent_category, class_name: 'Category', optional: true
```
Comment (igor04) [inline app/models/category.rb:6]: https://github.com/tux255/sports_hub_portal/pull/7#discussion_r658522044

File: app/views/admin/categories/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<% content_for(:add_item_button) { link_to 'Add Post', new_admin_post_path, class: 'btn btn-primary' } %>
+
+<div class="row py-3">
+  <h2><%= @title %></h2>
+  <div class="col-12">
+    <% @posts.each do |post| %>
+      <%= render partial: 'admin/shared/post', locals: {post: post} %>
+      <% end unless @posts.blank? %>
```
Comment (igor04) [inline app/views/admin/categories/index.html.erb:8]: https://github.com/tux255/sports_hub_portal/pull/7#discussion_r658530976

File: spec/factories/post.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        fixture_file_upload('test-600x400.png', 'image/png')
+      end
+    end
+
+    trait :with_no_content do
+      content { '' }
+    end
+
+    trait :with_no_title do
+      title { '' }
```
Comment (igor04) [inline spec/factories/post.rb:21]: https://github.com/tux255/sports_hub_portal/pull/7#discussion_r658534322

**Review by igor04 — COMMENTED** [review]
it looks like this PR also contains changes from already merged https://github.com/tux255/sports_hub_portal/pull/7 and comments from the previous PR still relevant

File: app/models/category.rb (hunk 2 above)
Comment (tux255) [reply app/models/category.rb:9]: solved by removing `, foreign_key: 'category_id'`

File: app/views/admin/categories/index.html.erb (hunk 4 above)
Comment (tux255) [reply app/views/admin/categories/index.html.erb:8]: it gives error if @posts are empty

**Comment by tux255** [comment]
> it looks like this PR also contains changes from already merged https://github.com/tux255/sports_hub_portal/pull/7 and comments from the previous PR still relevant

Yes it does contain. I am having issues with test and tried to fix it by merging back branches.

File: app/views/admin/categories/index.html.erb (hunk 4 above)
Comment (igor04) [reply app/views/admin/categories/index.html.erb:8]: what kind of error?

**Pull request merged**